packages/netlify-cms-backend-github/src/API.js

```diff
@@ -370,53 +468,33 @@ export default class API {
 
   editorialWorkflowGit(fileTree, entry, filesList, options) {
     const contentKey = entry.slug;
-    const branchName = this.generateBranchName(contentKey);
+    const oldBranchName = this.generateBranchName(contentKey);
+    let branchName = this.generateBranchName(
+      this.generateBaseName(options.collectionName, contentKey),
+    );
+    const newMeta = options.parsedData && options.parsedData.newMeta;
```

Comment:
Great to see support for both approaches to tracking statuses - the next step is to provide a bridge and make it optional. In the next major release, we drop the old approach, the bridge, and the option.

This is important because switching from metadata to labels, even if only for new entries, is a breaking change. If we make it optional, folks can opt in. I don't love this as it means most folks will continue using the old metadata until 3.0, but again, breaking is breaking. Consider the case where a repo already uses a labels with names that conflict with ours, for example.

The "bridge" concept means that, when enabled, this feature reads in metadata for existing entries, translates to our new approach, outputs a new label, and when the entry is successfully migrated, deletes the original metadata. This should probably happen to all entries at once when they're loaded.

The config option would probably be called `editorial_workflow_labels`, and would go under `backend`:

```yaml
backend:
  name: github
  repo: erquhart/test
  editorial_workflow_labels: true
```

Reply:
@erquhart  the only thing that would prevent the bridge option is inferring the collection name, with the new labels approach we can easily infer the collection name from the branch name. But with the currently metadata approach, there is no way to do that once we discard the metadata, and of course not possible to rename the branch. Can you think of any workarounds?

Reply:
Ah, hadn't noticed we were inferring the collection from the branch name - I actually think we can avoid storing metadata in the branch name altogether, and infer the collection name instead.

A collection inference function would operate on the list of files contained in a single PR:

1. If the PR has more than one file that is of a CMS editable type (md, json, toml, yaml), ignore the PR.
2. If the PR has no file that is of a CMS editable type, ignore the PR.
3. Compare with each collection config:
  1. First, file collections, use a direct string comparison of the file path against the collection's `file` config
  2. Second, folder collections, check the base path of the file against each folder collection's `folder` config
  3. If no match is found, ignore the PR
  4. If a match is found, use that collection's config to filter the file through the backend.js `filterEntries` method (perhaps via `processEntries`, not sure what's most efficient, it will depend on the implementation)
  5. If the file survives the filter, we have our collection name.

Notes:
* If a config allows a single file to show under multiple collections, that isn't a supported use case, so it's not a concern. In that case, the first matching collection will be inferred.
* The "ignore the PR" instances will often require at least a console error, as a `cms/` branch PR that doesn't match a collection really shouldn't be happening.

What do you think?

Reply:
Yes, can certainly infer the collection name by the steps you described above but found another blocker. Regarding migrating metadata to labels, I don't think we can't be able to delete the original metadata via the API (orphan refs). We may end up reading workflow status from two different sources.

Reply:
There's a method for deleting a reference: https://developer.github.com/v3/git/refs/#delete-a-reference

Did it not work for this? Worst case, we can delete the file at the ref and treat a ref with no file as migrated, but sending a DELETE request looks like it will work.

Reply:
> we can delete the file at the ref and treat a ref with no file as migrated

Yes that was my point, unlike the [get content](https://developer.github.com/v3/repos/contents/#get-contents) that has a ref param option, the [delete file](https://developer.github.com/v3/repos/contents/#delete-a-file) call doesn't.

Reply:
So what about deleting the ref via the `delete-a-reference` doc I linked?

Reply:
That would delete all metadata `json` files, so you are saying, after migrating we delete the ref?

Reply:
Right, migrate every entry and then delete the ref.